modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFClaimForm/utils/submitForm.ts

```diff
@@ -54,135 +56,135 @@ export default async function submitForm(
 					mdfRequest
 			  );
 
-	if (dtoMDFClaim?.id) {
-		const claimFolder = await createDocumentFolder(
-			claimParentFolderId,
-			`claim#${dtoMDFClaim.id}`
+	if (values.reimbursementInvoice && dtoMDFClaim?.id) {
+		const reimbursementInvoiceRenamed = renameFileKeepingExtention(
+			values.reimbursementInvoice,
+			`${values.reimbursementInvoice.name}#${dtoMDFClaim.id}`
 		);
 
-		if (values.reimbursementInvoice && claimFolder?.id) {
-			const claimDocument = await createDocumentFolderDocument(
-				claimFolder.id,
-				values.reimbursementInvoice
+		if (reimbursementInvoiceRenamed) {
+			const dtoReimbursementInvoice = await createDocumentFolderDocument(
+				claimParentFolderId,
+				reimbursementInvoiceRenamed
 			);
 
-			if (claimDocument?.contentUrl) {
-				await createMDFClaimDocuments(
-					values.reimbursementInvoice,
-					claimDocument.contentUrl,
-					dtoMDFClaim?.id
+			if (dtoReimbursementInvoice.id) {
+				await updateMDFClaim(
+					ResourceName.MDF_CLAIM_DXP,
+					values,
+					mdfRequest,
+					dtoMDFClaim.externalReferenceCodeSF,
+					dtoReimbursementInvoice.id,
+					dtoMDFClaim.id
 				);
 			}
 		}
+	}
 
-		if (values.activities?.length) {
-			const dtoMDFClaimActivities = await createMDFClaimActivities(
-				dtoMDFClaim.id,
-				values.activities.filter((activity) => activity.selected)
-			);
-
-			if (dtoMDFClaimActivities?.length) {
-				values.activities.map(async (activity, index) => {
-					const dtoActivity = dtoMDFClaimActivities[index];
-
-					if (dtoActivity?.id && claimFolder.id) {
-						const activityFolder = await createDocumentFolder(
-							claimFolder.id,
-							`activity#${dtoActivity.id}`
+	if (values.activities?.length && dtoMDFClaim?.id) {
+		await Promise.all(
+			values.activities
+				.filter((activity) => activity.selected)
+				?.map(async (activity) => {
```

Comment:
If you filtered, so the array exists `?` not necessary here

```suggestion
				.map(async (activity) => {
```

Reply:
The async worries me, this works??

map is synchronous, using the `async` and then `await` inside the filter and using PromiseAll makes me think if the response is being respected by the variables bellow.

I can take a look with you